src/components/ModalComponent.tsx

```diff
@@ -0,0 +1,17 @@
+import { ReactNode } from 'react';
+
+type ModalProps = {
+  isOpen: boolean;
+
+  children: ReactNode;
+};
+
+const ModalComponent = (props: ModalProps) => {
+  return (
+    <div className={props.isOpen ? 'modal-bg active' : 'modal-bg'}>
```

Comment:
for the handling class names you may use https://www.npmjs.com/package/classnames
please use it in the following pull requests :)